src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,483 @@
+    {
+        Block b = accessor.getBlock();
+        TileEntity te = accessor.getTileEntity();
+
+        if (b instanceof BlockOreTFC) currentTooltip = getOreTFCHead(stack, currentTooltip, accessor, config);
+        if (b instanceof BlockBarrel) currentTooltip = getBarrelHead(stack, currentTooltip, accessor, config);
+        if (te instanceof TECropBase) currentTooltip = getCropSimpleHead(stack, currentTooltip, accessor, config);
+        if (b instanceof BlockCropDead) currentTooltip = getCropDeadHead(stack, currentTooltip, accessor, config);
+        if (b instanceof BlockFruitTreeTrunk | b instanceof BlockFruitTreeLeaves | b instanceof BlockFruitTreeBranch)
+        {
+            currentTooltip = getFruitTreeHead(stack, currentTooltip, accessor, config);
+        }
+        return currentTooltip;
+    }
+
+    @Override
+    public List<String> getWailaBody(ItemStack stack, List<String> currentTooltip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        TileEntity te = accessor.getTileEntity();
+
+        // Mechanics
+        if (b instanceof BlockOreTFC) currentTooltip = getOreTFCBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEPlacedItemFlat)
+            currentTooltip = getPlacedItemFlatBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEPitKiln) currentTooltip = getPitKilnBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TECharcoalForge)
+            currentTooltip = getCharcoalForgeBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TELogPile) currentTooltip = getLogPileBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEBarrel) currentTooltip = getBarrelBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEBloomery) currentTooltip = getBloomeryBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEBlastFurnace)
+            currentTooltip = getBlastFurnaceBody(stack, currentTooltip, accessor, config);
+        else if (te instanceof TEFirePit) currentTooltip = getFirepitBody(stack, currentTooltip, accessor, config);
+            // Crops and Trees
+        else if (te instanceof TECropBase) currentTooltip = getCropSimpleBody(stack, currentTooltip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currentTooltip = getFruitTreeLeavesBody(stack, currentTooltip, accessor, config);
+        else if (b instanceof BlockBerryBush) currentTooltip = getBerryBushBody(stack, currentTooltip, accessor, config);
+        else if (b instanceof BlockCropDead) currentTooltip = getCropDeadBody(stack, currentTooltip, accessor, config);
+        else if (b instanceof BlockFruitTreeLeaves)
+            currentTooltip = getFruitTreeLeavesBody(stack, currentTooltip, accessor, config);
```

Comment:
Code style issue, which makes this harder to read and maintain later. (also, Alc already said this earlier)
Use this
```java
if (b instanceof BlockOreTFC)
{
    // do something
}
else if (b instanceof TEPlacedItemFlat)
{
    // do something else
}
else if...
```
instead.

But the optimal solution here is to code an interface here and put `getWailaBody()` and `getWailaHead()` there, then you just need to create classes for each `TileEntity` and/or `Block` that you made it does something when looked at and register everything in a `List<Interface> listWaila`, then the solution would be as clean as:
```java
    public List<String> getWailaBody(ItemStack stack, List<String> currentTooltip, IWailaDataAccessor accessor, IWailaConfigHandler config)
    {
        Block block = accessor.getBlock();
        TileEntity te = accessor.getTileEntity();
        for(<Interface> wailaInterface : listWaila)
        {
            if (waila.isLookingAt(b, te))
            {
                return waila.getBody()
            }
        }
    }

Reply:
I took option A. B may be optimal and if someone is eager to do it, please feel free